tests/actions.go

```diff
@@ -792,157 +817,29 @@ func (oa *operatorActions) UpgradeTidbClusterOrDie(info *TidbClusterConfig) {
-				}
-			}
-			return false, nil
-		} else if tc.Status.TiKV.Phase == v1alpha1.UpgradePhase {
-			glog.Infof("tikv is upgrading")
-			if tc.Status.TiDB.Phase == v1alpha1.UpgradePhase {
-				return false, pingcapErrors.New("tidb is upgrading while tikv is upgrading")
-			}
-			if !imageUpgraded(v1alpha1.PDMemberType, pdSet) {
-				return false, pingcapErrors.New("pd image is not updated while tikv is upgrading")
-			}
-			if !setUpgraded(pdSet) {
-				return false, pingcapErrors.New("pd stateful set is not upgraded while tikv is upgrading")
-			}
-			if !imageUpgraded(v1alpha1.TiKVMemberType, tikvSet) {
-				return false, pingcapErrors.New("tikv image is not updated while tikv is upgrading")
-			}
-			if !setUpgraded(tikvSet) {
-				if imageUpgraded(v1alpha1.TiDBMemberType, tidbSet) {
-					return false, pingcapErrors.New("tidb image is updated while tikv is upgrading")
-				}
-			}
-			return false, nil
-		} else if tc.Status.TiDB.Phase == v1alpha1.UpgradePhase {
-			glog.Infof("tidb is upgrading")
-			if !imageUpgraded(v1alpha1.PDMemberType, pdSet) {
-				return false, pingcapErrors.New("pd image is not updated while tidb is upgrading")
-			}
-			if !setUpgraded(pdSet) {
-				return false, pingcapErrors.New("pd stateful set is not upgraded while tidb is upgrading")
-			}
-			if !imageUpgraded(v1alpha1.TiKVMemberType, tikvSet) {
-				return false, pingcapErrors.New("tikv image is not updated while tidb is upgrading")
-			}
-			if !setUpgraded(tikvSet) {
-				return false, pingcapErrors.New("tikv stateful set is not upgraded while tidb is upgrading")
-			}
-			if !imageUpgraded(v1alpha1.TiDBMemberType, tidbSet) {
-				return false, pingcapErrors.New("tidb image is not updated while tikv is upgrading")
-			}
-			return false, nil
-		}
-
-		// check pd final state
-		if !imageUpgraded(v1alpha1.PDMemberType, pdSet) {
-			return false, nil
-		}
-		if !setUpgraded(pdSet) {
-			glog.Infof("check pd stateful set upgraded failed")
-			return false, nil
-		}
-		// check tikv final state
-		if !imageUpgraded(v1alpha1.TiKVMemberType, tikvSet) {
-			return false, nil
-		}
-		if !setUpgraded(tikvSet) {
-			glog.Infof("check tikv stateful set upgraded failed")
-			return false, nil
-		}
-		// check tidb final state
-		if !imageUpgraded(v1alpha1.TiDBMemberType, tidbSet) {
-			return false, nil
-		}
-		if !setUpgraded(tidbSet) {
-			glog.Infof("check tidb stateful set upgraded failed")
-			return false, nil
-		}
-		return true, nil
-	})
-}
-
 func (oa *operatorActions) DeployMonitor(info *TidbClusterConfig) error { return nil }
 func (oa *operatorActions) CleanMonitor(info *TidbClusterConfig) error  { return nil }
 
-func getComponentContainer(set *v1beta1.StatefulSet) (corev1.Container, bool) {
-	name := set.Labels[label.ComponentLabelKey]
-	for _, c := range set.Spec.Template.Spec.Containers {
-		if c.Name == name {
-			return c, true
+func getPodContainer(kubeCli kubernetes.Interface, namespace string, memberName string) (*corev1.Container, bool) {
```

Comment:
This function name is misleading, it retrieve the member container from member pod actually, so I think `getMemberContainer` is more descriptive

Reply:
done